libs/wingsdk/src/core/counter.ts

```diff
@@ -0,0 +1,59 @@
+import { IConstruct } from "constructs";
+import { App } from "./app";
+
+const COUNTERS_SYMBOL = Symbol.for("@winglang/sdk.core.Counter");
+
+/**
+ * A class that generates tree-unique sequenced IDs for constructs.
+ */
+export class Counters {
```

Comment:
The name of this obviously conflicts with the `Counter` resource, so I'm happy to hear some naming suggestions.

some other ideas:

IdGenerator
IdIncrementor
IdEnumerator

Reply:
i think we don't need an additional dedicated type for this. just plug it as a method of `App` and keep the state in there:

```js
App.of(this).makeId(scope, prefix)
```

Something like this...

Reply:
Is this important enough to add to the API of App?

I could also get rid of the class and make a global function `makeId(scope, prefix)` that gets the app of the scope and stores the data there the same way with a symbol.

Reply:
yes, i think it is. it could be useful for creating identifiers in user-defined classes. totally see this as a general facility of wing apps.

Reply:
Added